extension/chrome/elements/compose-modules/compose-storage-module.ts

```diff
@@ -75,28 +75,52 @@ export class ComposeStorageModule extends ViewModule<ComposeView> {
     // note by Tom 2021-08-10: We are only getting one public key from storage, but should
     //    work with arrays instead, like with `ContactStore.getOneWithAllPubkeys`.
     //    However, this whole fingerprint-base section seems unnecessary, we could likely
     //    remove it and the keys will be updated below using
     //    `lookupPubkeyFromKeyserversAndUpsertDb` anyway.
     //    discussion: https://github.com/FlowCrypt/flowcrypt-browser/pull/3898#discussion_r686229818
-    const [storedContact] = await ContactStore.get(undefined, [email]);
-    if (storedContact && storedContact.hasPgp && storedContact.pubkey && !storedContact.revoked) {
-      // checks if pubkey was updated, asynchronously. By the time user finishes composing,
-      //    newer version would have been updated in db.
-      // This implementation is imperfect in that, if sender didn't pull a particular pubkey
-      //    for a long time and the local pubkey has since expired, and there actually is a
-      //    newer version available on external key server, this may unnecessarily show "bad pubkey",
-      //    until next time user tries to enter recipient in the field again, which will at that point
-      //    get the updated key from db. This could be fixed by:
-      //      - either life fixing the UI after this call finishes, or
-      //      - making this call below synchronous and using the result directly
-      this.checkKeyserverForNewerVersionOfKnownPubkeyIfNeeded(storedContact).catch(Catch.reportErr);
-      return storedContact;
+    console.log(`>>>> Looking up in DB: ${email}`);
+    const storedContact = await ContactStore.getOneWithAllPubkeys(undefined, email);
+    console.log(">>>> " + (storedContact ? JSON.stringify(storedContact) : 'NOT_FOUND'));
+    if (storedContact && storedContact.sortedPubkeys.length) {
+      const results: PubKeyInfo[] = [];
+      for (const pubkey of storedContact.sortedPubkeys) {
+        if (!pubkey.revoked) {
+          // checks if pubkey was updated, asynchronously. By the time user finishes composing,
+          //    newer version would have been updated in db.
+          // This implementation is imperfect in that, if sender didn't pull a particular pubkey
+          //    for a long time and the local pubkey has since expired, and there actually is a
+          //    newer version available on external key server, this may unnecessarily show "bad pubkey",
+          //    until next time user tries to enter recipient in the field again, which will at that point
+          //    get the updated key from db. This could be fixed by:
+          //      - either life fixing the UI after this call finishes, or
+          //      - making this call below synchronous and using the result directly
+          this.checkKeyserverForNewerVersionOfKnownPubkeyIfNeeded(storedContact.info.email, pubkey)
+            .catch(Catch.reportErr);
+          results.push(pubkey);
+        } else {
```

Comment:
Now, with this loop, if have at least one revoked key, this method will wait for lookup completion, thus not providing immediate result. This is not needed if we have at least one valid key. If we have a valid key, we rightaway color the recipient green or yellow (expired but possible to encrypt) and do the `lookup` asynchronously

Reply:
@rrrooommmaaa I think I understood what is the problem, but I think absolutely don't understand how I should solve it,